tests/e2e/state-transition/chainsaw-test.yaml

```diff
@@ -0,0 +1,53 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: spark-operator-spark-application-state-transition-validation
+spec:
+  scenarios:
+  - bindings:
+      - name: TEST_NAME
+        value: succeeded
+      - name: FILE_NAME
+        value: spark-example-succeeded.yaml
+      - name: SPARK_APPLICATION_NAME
+        value: spark-job-succeeded-test
+  steps:
+  - try:
+    - script:
+        env:
+          - name: FILE_NAME
+            value: ($FILE_NAME)
+        content: kubectl apply -f $FILE_NAME
+    - assert:
+        timeout: 60s
+        file: "../assertions/spark-application/spark-state-transition.yaml"
+    catch:
+    - describe:
+        apiVersion: spark.apache.org/v1alpha1
+        kind: SparkApplication
+        namespace: default
+    finally:
+    - script:
+        env:
+          - name: SPARK_APPLICATION_NAME
+            value: ($SPARK_APPLICATION_NAME)
+        timeout: 30s
```

Comment:
Chainsaw default timeout time is 15 seconds: https://kyverno.github.io/chainsaw/latest/configuration/options/timeouts/?h=timeo#supported-timeouts